app/controllers/my_module_shareable_links_controller.rb

```diff
@@ -1,23 +1,65 @@
 # frozen_string_literal: true
 
 class MyModuleShareableLinksController < ApplicationController
-  before_action :load_my_module, except: %i(my_module_protocol_show)
+  before_action :load_my_module, except: %i(my_module_protocol_show
+                                            my_module_repositories_index_dt
+                                            my_module_repository_snapshots_index_dt)
   before_action :check_view_permissions, only: :show
-  before_action :check_manage_permissions, except: :my_module_protocol_show
-  skip_before_action :authenticate_user!, only: %i(my_module_protocol_show)
+  before_action :check_manage_permissions, except: %i(my_module_protocol_show
+                                                      my_module_repositories_index_dt
+                                                      my_module_repository_snapshots_index_dt)
+  before_action :shareable_link_load_my_module, only: %i(my_module_protocol_show
+                                                         my_module_repositories_index_dt
+                                                         my_module_repository_snapshots_index_dt)
+  before_action :load_repository, only: :my_module_repositories_index_dt
+  before_action :load_repository_snapshot, only: :my_module_repository_snapshots_index_dt
+  skip_before_action :authenticate_user!, only: %i(my_module_protocol_show
+                                                   my_module_repositories_index_dt
+                                                   my_module_repository_snapshots_index_dt)
   after_action -> { request.session_options[:skip] = true }
 
   def show
     render json: @my_module.shareable_link, serializer: ShareableLinksSerializer
   end
 
   def my_module_protocol_show
-    @shareable_link = ShareableLink.find_by(uuid: params[:uuid])
+    render 'shareable_links/my_module_protocol_show', layout: 'shareable_links'
+  end
 
-    return render_403 if @shareable_link.blank?
+  def my_module_repositories_index_dt
+    @draw = params[:draw].to_i
+    per_page = params[:length].to_i < 1 ? Constants::REPOSITORY_DEFAULT_PAGE_SIZE : params[:length].to_i
+    page = (params[:start].to_i / per_page) + 1
+    datatable_service = RepositoryDatatableService.new(@repository, params, nil, @my_module)
 
-    @my_module = @shareable_link.shareable
-    render 'shareable_links/my_module_protocol_show', layout: 'shareable_links'
+    @datatable_params = {
+      view_mode: params[:view_mode],
+      my_module: @my_module,
+      include_stock_consumption: @repository.has_stock_management? && params[:assigned].present?,
+      disable_stock_management: true # stock management is always disabled in MyModule context
+    }
+
+    @all_rows_count = datatable_service.all_count
+    @columns_mappings = datatable_service.mappings
+
+    @repository_rows = datatable_service.repository_rows.page(page).per(per_page)
+
+    render 'repository_rows/simple_view_index'
+  end
+
+  def my_module_repository_snapshots_index_dt
```

Comment:
don't use my_module_ prefix and it should be singular: repository_snapshot_index_dt